avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -691,9 +691,6 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
       case STRING:
         if (enhancedSchemaSupport && schema.parameters() != null
             && schema.parameters().containsKey(AVRO_TYPE_ENUM)) {
-          String enumSchemaName = schema.parameters().get(AVRO_TYPE_ENUM);
-          org.apache.avro.Schema cachedEnum = schemaMap.get(enumSchemaName);
-          if (cachedEnum == null) {
```

Comment:
I'm not sure I understand your reasoning here about how it should already be cached... How would this be guaranteed?